core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -264,10 +264,10 @@ private[spark] class ExecutorAllocationManager(
     updateAndSyncNumExecutorsTarget(now)
 
     removeTimes.retain { case (executorId, expireTime) =>
-      val expired = now >= expireTime
+      var expired = now >= expireTime
       if (expired) {
-        initializing = false
-        removeExecutor(executorId)
+        expired = removeExecutor(executorId)
+        if (expired)  initializing = false
```

Comment:
Without commenting on the validity of the change, you have some style problems, like this needing to be in a block in braces